src/unification.hpp

```diff
@@ -31,7 +31,7 @@ struct Core {
 	void unify(int i, int j);
 
 	int new_var(const char* debug = nullptr);
-	int new_term(int f, std::vector<int> args, const char* debug = nullptr);
+	int new_term(int f = -1, std::vector<int> args = {}, const char* debug = nullptr);
```

Comment:
remove the default -1 argument. That is not the common usage of this function. We would normally want a compilation error if no arguments are passed.

Reply:
Figured it'd be bad, I wanted to try a method in which we don't have to add the comment explaining why we call with -1 on metatype term creation. I will add it tho